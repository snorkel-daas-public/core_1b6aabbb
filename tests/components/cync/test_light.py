"""Tests for the Cync integration light platform."""

from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import setup_integration

from tests.common import MockConfigEntry, snapshot_platform


async def test_entities(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test that light attributes are properly set on setup."""

    await setup_integration(hass, mock_config_entry)

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
