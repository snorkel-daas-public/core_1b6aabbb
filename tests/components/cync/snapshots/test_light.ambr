# serializer version: 1
# name: test_entities[light.bedroom_lamp-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max_color_temp_kelvin': 7000,
      'max_mireds': 500,
      'min_color_temp_kelvin': 2000,
      'min_mireds': 142,
      'supported_color_modes': list([
        <ColorMode.COLOR_TEMP: 'color_temp'>,
        <ColorMode.RGB: 'rgb'>,
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'light',
    'entity_category': None,
    'entity_id': 'light.bedroom_lamp',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': None,
    'platform': 'cync',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'light',
    'unique_id': '1000-1101',
    'unit_of_measurement': None,
  })
# ---
# name: test_entities[light.bedroom_lamp-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'brightness': 205,
      'color_mode': <ColorMode.COLOR_TEMP: 'color_temp'>,
      'color_temp': 333,
      'color_temp_kelvin': 2999,
      'friendly_name': 'Bedroom Lamp',
      'hs_color': tuple(
        27.827,
        56.922,
      ),
      'max_color_temp_kelvin': 7000,
      'max_mireds': 500,
      'min_color_temp_kelvin': 2000,
      'min_mireds': 142,
      'rgb_color': tuple(
        255,
        177,
        110,
      ),
      'supported_color_modes': list([
        <ColorMode.COLOR_TEMP: 'color_temp'>,
        <ColorMode.RGB: 'rgb'>,
      ]),
      'supported_features': <LightEntityFeature: 0>,
      'xy_color': tuple(
        0.496,
        0.383,
      ),
    }),
    'context': <ANY>,
    'entity_id': 'light.bedroom_lamp',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entities[light.lamp_bulb_1-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max_color_temp_kelvin': 7000,
      'max_mireds': 500,
      'min_color_temp_kelvin': 2000,
      'min_mireds': 142,
      'supported_color_modes': list([
        <ColorMode.COLOR_TEMP: 'color_temp'>,
        <ColorMode.RGB: 'rgb'>,
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'light',
    'entity_category': None,
    'entity_id': 'light.lamp_bulb_1',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': None,
    'platform': 'cync',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'light',
    'unique_id': '1000-1111',
    'unit_of_measurement': None,
  })
# ---
# name: test_entities[light.lamp_bulb_1-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'brightness': 230,
      'color_mode': <ColorMode.RGB: 'rgb'>,
      'color_temp': None,
      'color_temp_kelvin': None,
      'friendly_name': 'Lamp Bulb 1',
      'hs_color': tuple(
        215.0,
        33.333,
      ),
      'max_color_temp_kelvin': 7000,
      'max_mireds': 500,
      'min_color_temp_kelvin': 2000,
      'min_mireds': 142,
      'rgb_color': tuple(
        120,
        145,
        180,
      ),
      'supported_color_modes': list([
        <ColorMode.COLOR_TEMP: 'color_temp'>,
        <ColorMode.RGB: 'rgb'>,
      ]),
      'supported_features': <LightEntityFeature: 0>,
      'xy_color': tuple(
        0.248,
        0.27,
      ),
    }),
    'context': <ANY>,
    'entity_id': 'light.lamp_bulb_1',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entities[light.lamp_bulb_2-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max_color_temp_kelvin': 7000,
      'max_mireds': 500,
      'min_color_temp_kelvin': 2000,
      'min_mireds': 142,
      'supported_color_modes': list([
        <ColorMode.COLOR_TEMP: 'color_temp'>,
        <ColorMode.RGB: 'rgb'>,
      ]),
    }),
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'light',
    'entity_category': None,
    'entity_id': 'light.lamp_bulb_2',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': None,
    'platform': 'cync',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'light',
    'unique_id': '1000-1112',
    'unit_of_measurement': None,
  })
# ---
# name: test_entities[light.lamp_bulb_2-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Lamp Bulb 2',
      'max_color_temp_kelvin': 7000,
      'max_mireds': 500,
      'min_color_temp_kelvin': 2000,
      'min_mireds': 142,
      'supported_color_modes': list([
        <ColorMode.COLOR_TEMP: 'color_temp'>,
        <ColorMode.RGB: 'rgb'>,
      ]),
      'supported_features': <LightEntityFeature: 0>,
    }),
    'context': <ANY>,
    'entity_id': 'light.lamp_bulb_2',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
